src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -125,6 +126,15 @@ internal static SafeSslHandle AllocateSslContext(SslProtocols protocols, SafeX50
                         throw CreateSslException(SR.net_allocate_ssl_context_failed);
                     }
 
+                    if (!sslAuthenticationOptions.IsServer)
+                    {
+                        // The IdnMapping converts unicode input into the IDNA punycode sequence.
+                        string punyCode = new IdnMapping().GetAscii(sslAuthenticationOptions.TargetHost);
```

Comment:
I second Stephen here. does this behavior match whatever standard we are using? where did we pick this behavior? documenting this is important as we document all used protocols in general.

other point, IdnMapping can throw exceptions. is this ok? does this can change the current behavior?

---
In reply to: [150443017](https://github.com/dotnet/corefx/pull/25118#discussion_r150443017) [](ancestors = 150443017)